File: trunk/ShadowFrameworkLite20/src/shadow/pipeline/java/SFPrimitiveProgramAssociation.java
package shadow.pipeline.java;

import shadow.pipeline.SFProgramComponent;
import shadow.pipeline.parameters.SFPipelineRegister;

public class SFPrimitiveProgramAssociation {
	
	private SFPipelineRegister register;
	private SFProgramComponent program;
	
	public SFPrimitiveProgramAssociation(SFPipelineRegister register, SFProgramComponent program) {
		super();
		this.register = register;
		this.program = program;
	}
	
	public SFPipelineRegister getRegister() {
		return register;
	}
	
	public SFProgramComponent getProgram() {
		return program;
	}

}

File: trunk/SFDataApplicationTestC++/shadow/pipeline/OpenGL20/SFPrimitiveProgramAssociation.cpp

#include "SFPrimitiveProgramAssociation.h"


namespace sf{
SFPrimitiveProgramAssociation::SFPrimitiveProgramAssociation(SFPipelineRegister* reg,
                                  SFProgramComponent* program) {
	this->reg = reg;
	this->program = program;
}

SFPipelineRegister* SFPrimitiveProgramAssociation::getRegister() {
	return reg;
}

SFProgramComponent* SFPrimitiveProgramAssociation::getProgram() {
	return program;
}
}
